api/v1beta1/metal3cluster_webhook.go

```diff
@@ -68,6 +74,79 @@ func (c *Metal3Cluster) validate() error {
 		)
 	}
 
+	if c.Spec.CloudProviderEnabled != nil && c.Spec.NoCloudProvider != nil {
+		if !*c.Spec.CloudProviderEnabled && !*oldM3C.Spec.NoCloudProvider {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "cloudProviderEnabled"),
+					c.Spec.CloudProviderEnabled,
+					"cloudProviderEnabled conflicts the value of noCloudProvider",
+				),
+			)
+		}
+		if *c.Spec.CloudProviderEnabled && *oldM3C.Spec.NoCloudProvider {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					field.NewPath("spec", "cloudProviderEnabled"),
+					c.Spec.CloudProviderEnabled,
+					"cloudProviderEnabled conflicts the value of noCloudProvider",
+				),
+			)
+		}
+	}
+
+	if oldM3C != nil {
```

Comment:
Isn't it the same as the previous check?

Reply:
now it's different after the last push.  Please check
So basically this check is when we already have some value set of one variable(CloudProviderEnabled or NoCloudProvider) and the user is changing the other one, and the previous check is when the user is changing both CloudProviderEnabled and NoCloudProvider at the same time.